src/Shortcodes/FileShortcodeProvider.php

```diff
@@ -136,17 +136,30 @@ public static function handle_shortcode($arguments, $content, $parser, $shortcod
      */
     protected static function getCachedMarkup($cache, $cacheKey, $arguments): string
     {
+        // Retrieve the cache markup
         $item = $cache->get($cacheKey);
+        if (empty($item) || empty($item['markup'])) {
+            // This is a cache miss
+            return '';
+        }
+
+        // Check the file exists
+        $filename = $item['filename'];
+        $hash = $item['hash'];
         $assetStore = Injector::inst()->get(AssetStore::class);
-        if ($item && $item['markup'] && !empty($item['filename'])) {
-            // Initiate a protected asset grant if necessary
-            $allowSessionGrant = static::getGrant(null, $arguments);
-            if ($allowSessionGrant && $assetStore->exists($item['filename'], $item['hash'])) {
```

Comment:
`$allowSessionGrant` will be false for most project. So we basically never return the markup.

Checking the file exists is fundamentally a different concern, so I split this up in the updated implementation.